pyqg/layered_model.py

```diff
@@ -118,8 +118,8 @@ def __init__(
         if H is None: H = [500] + [1750 for _ in range(nz-1)]
         if rho is None: rho = np.arange(nz) * 0.3 + 1025
 
-        self.Ubg = np.array(U)
-        self.Vbg = np.array(V)
+        self._U = U
+        self._V = V
```

Comment:
Can you explain why you moved this?

Reply:
This needed to get moved up because we can't set `Ubg` before calling `super`, since `Ubg` is now a property of the kernel. However, `super` calls `_initialize_background` which then calls `_initialize_stretching_matrix` which requires `Ubg` to be defined. So, following `QGModel`, I moved the actual setting of `Ubg` to `_initialize_background` (which makes sense). The issue is that we need some intermediate variable to store the array.